### PR TITLE
add -f to plugin inspect

### PR DESCRIPTION
**Pull request by vieux**
depends on https://github.com/docker/engine-api/pull/376

ping @anusha-ragunathan @tiborvass

### CODE REVIEW

**Comment by vdemeester** [comment]
Design LTGM :angel:

File: api/client/plugin/inspect.go [hunk 1]
```diff
@@ -3,50 +3,57 @@
...
+	format      string
+}
+
 func newInspectCommand(dockerCli *client.DockerCli) *cobra.Command {
+	var opts inspectOptions
+
 	cmd := &cobra.Command{
 		Use:   "inspect PLUGIN",
 		Short: "Inspect a plugin",
 		Args:  cli.ExactArgs(1),
```
Comment (anusha-ragunathan) [inline api/client/plugin/inspect.go]: pluginNames is an array, so this should be `cli.MinArgs(1)`

File: vendor/src/github.com/docker/engine-api/client/plugin_inspect.go [hunk 2]
```diff
@@ -3,20 +3,28 @@
...
 import (
+	"bytes"
 	"encoding/json"
+	"io/ioutil"
 
 	"github.com/docker/engine-api/types"
 	"golang.org/x/net/context"
 )
 
 // PluginInspect inspects an existing plugin
```
Comment (anusha-ragunathan) [inline vendor/src/github.com/docker/engine-api/client/plugin_inspect.go]: Update function name in comment

File: docs/reference/commandline/plugin_inspect.md [hunk 3]
```diff
@@ -17,7 +17,8 @@ Usage:  docker plugin inspect PLUGIN
 Inspect a plugin
 
 Options:
-      --help   Print usage
+      -f, --format string   Format the output using the given go template
```
Comment (anusha-ragunathan) [inline docs/reference/commandline/plugin_inspect.md]: Can you add an example with format?

**Comment by anusha-ragunathan** [comment]
@vieux : Please update this test while you are here. https://github.com/docker/docker/blob/master/integration-cli/docker_cli_plugins_test.go#L40

File: api/client/plugin/inspect.go (hunk 1 above)
Comment (vieux) [inline api/client/plugin/inspect.go]: right, thanks

**Comment by vieux** [comment]
depends on https://github.com/docker/docker/pull/25424

File: docs/reference/commandline/plugin_inspect.md (hunk 3 above)
Comment (vieux) [inline docs/reference/commandline/plugin_inspect.md]: done

**Comment by vieux** [comment]
@anusha-ragunathan I think I addressed all your comments.

**Comment by anusha-ragunathan** [comment]
LGTM

**Comment by icecrime** [comment]
LGTM

**Comment by vdemeester** [comment]
docs LGTM :cat:

**Pull request merged**